paddle/fluid/framework/data_layout_transform.cc

```diff
@@ -88,5 +91,84 @@ void TransDataLayout(const OpKernelType& kernel_type_for_var,
   out->set_layout(expected_kernel_type.data_layout_);
 }
 
+#ifdef PADDLE_WITH_MKLDNN
+using mkldnn::memory;
+using mkldnn::primitive;
+using mkldnn::reorder;
+
+void* get_data_from_tensor(const Tensor& tensor,
+                           mkldnn::memory::data_type type) {
+  switch (type) {
+    case mkldnn::memory::data_type::f32:
+      return platform::to_void_cast(tensor.data<float>());
+    case mkldnn::memory::data_type::s8:
+      return platform::to_void_cast(tensor.data<char>());
+    case mkldnn::memory::data_type::u8:
+      return platform::to_void_cast(tensor.data<unsigned char>());
+    case mkldnn::memory::data_type::s16:
+      return platform::to_void_cast(tensor.data<int16_t>());
+    case mkldnn::memory::data_type::s32:
+      return platform::to_void_cast(tensor.data<int32_t>());
+    default:
+      PADDLE_THROW("wrong mkldnn type provided");
+  }
+}
+
+void TransDataLayoutMkldnn(const OpKernelType& kernel_type_for_var,
```

Comment:
How about `TransDataLayoutMkldnn` be `TransDataLayoutMKLDNN`? Since @wangkuiyi suggest in https://github.com/PaddlePaddle/Paddle/pull/3337#discussion_r131963133

Reply:
Done.